comgt: ncm.sh manufacturer bug remove

stramoslab opened the pull request:
Signed-off-by: Alexander Stramoslab sicorski@madus.org

Maintainer: Felix Fietkau / <nbd@nbd.name>
Compile tested: GenuineIntel i-5 9400F, master
Run tested: YK-L1c, modem huawei E3276, OpenWrt 19.07.5 r11257-5090152ae3

Description:
[FIX] Fixed the appearance of annoying messages instead of the manufacturer's name.

aparcar commented:
Hi, thanks for your contribution!

Please read through [the patch submitting guidelines](https://openwrt.org/submitting-patches). Your patch patch has some formal issues.

Most commonly are the following:
* Prefix the commit subject with the affected *area*, find that area by looking at `git log <modified_file>`
* Add a commit message describing why this patch was required
* Add a Signed-of-by line containing your real name

Commit:
comgt-ncm: fixed appearance of annoying messages instead of the
               manufacturer name

    Some modems like huawei e3276 may generate status messages what
    starts from ^ symbol. While ncm.sh script try determine
    manufacturer name, this strings can catch like manufacturer name.
    This changes remove all strings what starts from ^ symbol from
    manufacturer request.

    Signed-off-by: Alexander Stramoslab <sicorski@madus.org>

stramoslab commented:
I hope I did everything right this time

adschm commented:
@aparcar Why me? I've never been able to understand awk no matter how hard I tried....

aparcar commented:
> @aparcar Why me? I've never been able to understand awk no matter how hard I tried....

You're doing pretty well with the rest so I thought it's time for some new challenges

stramoslab commented:
> 
> 
> @aparcar Why me? I've never been able to understand awk no matter how hard I tried....
I can help you understand what is going on in these lines

CodeFetch commented:
@stramoslab Can you please describe what your patch is doing?
Because I thought this was fixed in https://github.com/openwrt/openwrt/pull/3231 and was due to the modem not being in an operable state, yet.

CodeFetch commented:
@stramoslab I mean... Does it actually fix this initialization delay issue?

stramoslab commented:
This actually fixes another bug.During operation, the modem sends information about the current signal level to the console. It looks something like this:
^RSSI:22
^LTERSRP:-101,-14
^RSSI:23
^LTERSRP:-101,-17
Sometimes the output of these lines coincides in time with the request for the manufacturer's name, and all this mess of lines is returned by the script "gcom -d "$device" -s /etc/gcom/getcardinfo.gcom"
Then awk throws out the excess, and returns the first non-empty string, which may be, for example "^RSSI:23" instead of "huawei".
the addition I suggested will additionally throw out lines starting with the character ^. after that, the first non-empty string is doomed to be only manufacturer name.
fix #3231 is about something else, it tries to find out the manufacturer's name again if the modem didn't respond. In my case, the modem responds, but also writes service information that needs to be cut off.
